Jint/Native/JsValue.cs

```diff
@@ -168,8 +168,9 @@ internal IIterator GetIterator(Engine engine)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal bool TryGetIterator(Engine engine, out IIterator iterator)
         {
-            if (!(this is ObjectInstance oi)
-                || !oi.TryGetValue(GlobalSymbolRegistry.Iterator._value, out var value)
+            var objectInstance = TypeConverter.ToObject(engine, this);
+
+            if (!objectInstance.TryGetValue(GlobalSymbolRegistry.Iterator._value, out var value)
```

Comment:
should Iterator._value be internal and look like `Iterator.Value` if that's the expected usage?

Reply:
That might make it look a bit prettier, now it's just direct field access.